src/matrix.js

```diff
@@ -24,53 +24,150 @@ SVG.Matrix = SVG.invent({
+      if (isFinite(px) && isFinite(py)) {
+        // Figure out where the origin went and the delta to get there
+        var p = new SVG.Point(ox - rx, oy - ry).transform(transformer)
+        var dx = px - p.x
+        var dy = py - p.y
+
+        // Apply another translation
+        transformer = transformer.translate(dx, dy)
+      }
+
+      // We can apply translations after everything else
+      transformer = transformer.translate(tx, ty)
+      return transformer
+    },
+
+    // Applies a matrix defined by its affine parameters
+    compose: function (o) {
+      // Get the parameters
+      var sx = o.scaleX || 1
+      var sy = o.scaleY || 1
+      var lam = o.shear || 0
+      var theta = o.rotate || 0
+      var tx = o.translateX || 0
+      var ty = o.translateY || 0
+
+      // Apply the standard matrix
+      var result = new SVG.Matrix()
+        .scale(sx, sy)
+        .shear(lam)
+        .rotate(theta)
+        .translate(tx, ty)
+        .lmultiply(this)
+      return result
+    },
+
+    // Decomposes this matrix into its affine parameters
+    decompose: function () {
+      // Get the parameters from the matrix
+      var a = this.a
+      var b = this.b
+      var c = this.c
+      var d = this.d
+      var e = this.e
+      var f = this.f
+
+      // Figure out if the winding direction is clockwise or counterclockwise
+      var determinant = a * d - b * c
+      var ccw = determinant > 0 ? 1 : -1
+
+      // Since we only shear in x, we can use the x basis to get the x scale
+      // and the rotation of the resulting matrix
+      var sx = ccw * Math.sqrt(a * a + b * b)
+      var theta = 180 / Math.PI * Math.atan2(ccw * b, ccw * a)
+
+      // We can then solve the y basis vector simultaneously to get the other
+      // two affine parameters directly from these parameters
+      var lam = (a * c + b * d) / determinant
+      var sy = ((c * sx) / (lam * a - b)) || ((d * sx) / (lam * b + a))
+
+      // Construct the decomposition and return it
       return {
-        // translation
-        x: this.e,
-        y: this.f,
-        transformedX: (this.e * Math.cos(skewX * Math.PI / 180) + this.f * Math.sin(skewX * Math.PI / 180)) / Math.sqrt(this.a * this.a + this.b * this.b),
-        transformedY: (this.f * Math.cos(skewX * Math.PI / 180) + this.e * Math.sin(-skewX * Math.PI / 180)) / Math.sqrt(this.c * this.c + this.d * this.d),
-        // skew
-        skewX: -skewX,
-        skewY: 180 / Math.PI * Math.atan2(py.y, py.x),
-        // scale
-        scaleX: Math.sqrt(this.a * this.a + this.b * this.b),
-        scaleY: Math.sqrt(this.c * this.c + this.d * this.d),
-        // rotation
-        rotation: skewX,
+        // Return the affine parameters
+        scaleX: sx,
+        scaleY: sy,
+        shear: lam,
+        rotate: theta,
+        translateX: e,
+        translateY: f,
+
+        // Return the matrix parameters
+        matrix: new SVG.Matrix(this),
```

Comment:
I actually think we can skip this one when we have a-f in this object anyway. The object you return back is already a matrix-like object because of a-f. So you can pass it in where a matrix is needed directly. And if you want a clean matrix you can just do `new SVG.Matrix(el.transform())`

Reply:
Good point. Done!